apps/dashboard/app/models/project.rb

```diff
@@ -27,33 +28,69 @@ def find(project_path)
     def dataroot
       Rails.logger.debug("project path is: #{OodAppkit.dataroot.join('projects')}")
 
-      OodAppkit.dataroot.join('projects').tap do |p|
-        p.mkpath unless p.exist?
+      OodAppkit.dataroot.join('projects').tap do |path|
+        path.mkpath unless path.exist?
       rescue StandardError => e
         Pathname.new('')
       end
     end
   end
 
+  validates :dir, presence: true
+  validates :dir, format: {
+
+    with: /\A[\w-]+\z/,
+    message: 'Directory may only contain letters, digits, dashes, and underscores'
+  }
+
   attr_reader :dir
+  delegate :icon, :name, :description, to: :manifest
+
+  def initialize(attributes = {})
+    @dir            = attributes.fetch(:dir, nil).to_s
+  end
 
-  def initialize(args = {})
-    @dir = args.fetch(:dir, nil).to_s
+  def save
+    write_manifest
+  rescue => error
+    errors.add(:save, error.message)
+    Rails.logger.error("ERROR: #{error.class} - #{error.message}")
+    false
```

Comment:
Maybe if we add a `to_h` (or maybe ActiveModel already gives it) we can reuse `update`.

Beyond that, I don't believe there are errors to rescue (manifest rescues them all), but we can capture that we didn't save the manifest correctly.
```suggestion
    res = update(to_h)
    errors.add(:save, 'Cannot save manifest') unless res
    res
```